### PR TITLE
items: fix requested loans by library list

### PR DESCRIPTION
**Pull request by zannkukai**
Closes rero/rero-ils#2296.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: rero_ils/modules/loans/api.py [hunk 1]
```diff
@@ -319,85 +319,98 @@ def loans_pending():
...
+        def library_name_by_pid(pid):
             """Get library name by pid.
 
-            :param library_pid: the library pid.
+            :param pid: the library pid.
             """
-            if library_pid not in libraries:
-                libraries[library_pid] = LibrariesSearch()\
-                    .filter('term', pid=library_pid)\
+            if pid not in libraries:
```
Comment (rerowep) [inline rero_ils/modules/loans/api.py:360]: See above (`libraries`)

File: rero_ils/modules/loans/api.py [hunk 2]
```diff
@@ -319,85 +319,98 @@ def loans_pending():
...
             """Get holdings by pid.
 
-            :param holding_pid: the holdings pid.
+            :param pid: the holdings pid.
             """
             from ..holdings.api import HoldingsSearch
-            if holding_pid not in holdings:
-                holdings[holding_pid] = HoldingsSearch()\
-                    .filter('term', pid=library_pid)\
+            if pid not in holdings:
```
Comment (rerowep) [inline rero_ils/modules/loans/api.py:376]: See above (`holdings`)

File: rero_ils/modules/loans/api.py [hunk 3]
```diff
@@ -319,85 +319,98 @@ def loans_pending():
...
 
-            :param item_pid: the item pid.
+            :param pid: the item pid.
             """
-            if item_pid not in items:
-                items[item_pid] = ItemsSearch()\
-                    .filter('term', pid=item_pid)\
+            fields = ['pid', 'barcode', 'call_number', 'library', 'location',
+                      'temporary_item_type', 'holding']
+            if pid not in items:
```
Comment (rerowep) [inline rero_ils/modules/loans/api.py:393]: See above (`items`)

File: rero_ils/modules/loans/api.py [hunk 4]
```diff
@@ -319,85 +319,98 @@ def loans_pending():
...
             """Get item type by pid.
 
-            :param item_pid: the item_type pid.
+            :param pid: the item_type pid.
             """
-            if item_type_pid not in item_types:
-                item_types[item_type_pid] = ItemTypesSearch()\
-                    .filter('term', pid=item_type_pid)\
-                    .filter('term', negative_availability=True)\
+            if pid not in item_types:
```
Comment (rerowep) [inline rero_ils/modules/loans/api.py:407]: See above (`item_types`)

**Review by rerowep — CHANGES_REQUESTED** [review]

File: rero_ils/modules/items/views/api_views.py [hunk 5]
```diff
@@ -79,9 +79,9 @@ def decorated_view(*args, **kwargs):
         try:
             return func(*args, **kwargs)
         except NotFound as error:
-            raise(error)
```
[the remark this answers is not in the document]
Comment (zannkukai) [reply rero_ils/modules/items/views/api_views.py:82]: @rerowep
I'm not agree. What's you propose is to build a new `NotFound` exception based on another `NotFound` exception.
Seems the same than write `raise NotFound(NotFound('my exception message'))`.

the python `raise` statement needs an exception as argument. In this case, the `error` variable is already a `NotFound(Exception)` exception. https://stackoverflow.com/questions/18188563/how-to-re-raise-an-exception-in-nested-try-except-blocks?answertab=votes#tab-top

**Review by rerowep — APPROVED** [review]

**Review by BadrAly — APPROVED** [review]

**Pull request merged**